[CI] ConcurrentSeqNoVersioningIT testSeqNoCASLinearizability failing

**Build Scans:**
- [elasticsearch-intake #37714 / part1](https://gradle-enterprise.elastic.co/s/o7bnh7iw6xr2c)

**Reproduction Line:**
```
./gradlew ":server:internalClusterTest" --tests "org.elasticsearch.versioning.ConcurrentSeqNoVersioningIT.testSeqNoCASLinearizability" -Dtests.seed=4B031025F6FF17F8 -Dtests.locale=bez-TZ -Dtests.timezone=America/Managua -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.versioning.ConcurrentSeqNoVersioningIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testSeqNoCASLinearizability'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: safeGet: listener was not completed within the timeout
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testSeqNoCASLinearizability
- [main] 1 consecutive failures in step part1
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

During post-test cleanup and validation:
```
WARNING: Uncaught exception in thread: Thread[#10779,elasticsearch[node_t2][management][T#1],5,TGRP-ConcurrentSeqNoVersioningIT]	
java.lang.AssertionError: Unable to find master in cluster state. Expecting a stable master node	
	at __randomizedtesting.SeedInfo.seed([4B031025F6FF17F8]:0)	
	at org.elasticsearch.test.ESIntegTestCase.lambda$doEnsureClusterStateConsistency$27(ESIntegTestCase.java:1763)	
	at org.elasticsearch.action.support.SubscribableListener.lambda$andThenAccept$3(SubscribableListener.java:601)	
	at org.elasticsearch.action.ActionListenerImplementations$MappedActionListener.onResponse(ActionListenerImplementations.java:105)	
	at org.elasticsearch.action.support.SubscribableListener$SuccessResult.complete(SubscribableListener.java:436)	
	at org.elasticsearch.action.support.SubscribableListener.tryComplete(SubscribableListener.java:347)	
	at org.elasticsearch.action.support.SubscribableListener.setResult(SubscribableListener.java:376)	
	at org.elasticsearch.action.support.SubscribableListener.onResponse(SubscribableListener.java:283)	
	at org.elasticsearch.action.ActionListener$3.onResponse(ActionListener.java:432)	
	at org.elasticsearch.tasks.TaskManager$1.onResponse(TaskManager.java:223)	
	at org.elasticsearch.tasks.TaskManager$1.onResponse(TaskManager.java:217)	
	at org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onResponse(ActionListenerImplementations.java:350)	
	at org.elasticsearch.action.ActionListener$3.onResponse(ActionListener.java:432)	
	at org.elasticsearch.action.ActionListener.completeWith(ActionListener.java:392)	
	at org.elasticsearch.action.admin.cluster.state.TransportClusterStateAction.localClusterStateOperation(TransportClusterStateAction.java:120)	
	at org.elasticsearch.action.admin.cluster.state.TransportClusterStateAction.localClusterStateOperation(TransportClusterStateAction.java:59)	
	at org.elasticsearch.action.support.local.TransportLocalClusterStateAction.lambda$innerDoExecute$0(TransportLocalClusterStateAction.java:92)	
	at org.elasticsearch.action.ActionRunnable$4.doRun(ActionRunnable.java:101)	
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1114)	
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)	
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)	
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)	
	at java.base/java.lang.Thread.run(Thread.java:1516)
```
looks like t2 was the master but stepped down at the end of the last round probably as a side effect of partition healing
```
[2026-04-16T11:16:51,411][INFO ][o.e.t.InternalTestCluster][testSeqNoCASLinearizability] Clearing active scheme network disruption (disruption type: network unresponsive, disrupted links: two partitions (partition 1: [node_t8, node_t6, node_t5, node_t4, node_t3, node_t1, node_t0] and partition 2: [node_t7, node_t2])), expected healing time 0s	
[2026-04-16T11:16:51,411][INFO ][o.e.t.d.NetworkDisruption][testSeqNoCASLinearizability] stop disrupting (disruption scheme: network unresponsive, disrupted links: two partitions (partition 1: [node_t8, node_t6, node_t5, node_t4, node_t3, node_t1, node_t0] and partition 2: [node_t7, node_t2]))	
[2026-04-16T11:16:51,412][INFO ][o.e.c.c.Coordinator      ][node_t0][cluster_coordination][T#1] master node [{node_t2}{tN0GU0iORmeSB6p8dVJ6nA}{H2TVjHoKTaO87YM9yTDKQg}{node_t2}{127.0.0.1}{127.0.0.1:25065}{ilmrt}{9.5.0}{8000099-9095000}] disconnected, restarting discovery [[node_t2][127.0.0.1:25065][internal:coordination/fault_detection/leader_check] disconnected]	
[2026-04-16T11:16:51,412][INFO ][o.e.t.ClusterConnectionManager][testSeqNoCASLinearizability] transport connection to [{node_t2}{tN0GU0iORmeSB6p8dVJ6nA}{H2TVjHoKTaO87YM9yTDKQg}{node_t2}{127.0.0.1}{127.0.0.1:25065}{ilmrt}{9.5.0}{8000099-9095000}] closed by remote; if unexpected, see [https://www.elastic.co/docs/troubleshoot/elasticsearch/troubleshooting-unstable-cluster?version=master#troubleshooting-unstable-cluster-network] for troubleshooting guidance	
[2026-04-16T11:16:51,413][INFO ][o.e.t.ClusterConnectionManager][testSeqNoCASLinearizability] transport connection to [{node_t2}{tN0GU0iORmeSB6p8dVJ6nA}{H2TVjHoKTaO87YM9yTDKQg}{node_t2}{127.0.0.1}{127.0.0.1:25065}{ilmrt}{9.5.0}{8000099-9095000}] closed by remote; if unexpected, see [https://www.elastic.co/docs/troubleshoot/elasticsearch/troubleshooting-unstable-cluster?version=master#troubleshooting-unstable-cluster-network] for troubleshooting guidance	
[2026-04-16T11:16:51,413][INFO ][o.e.c.c.Coordinator      ][node_t1][cluster_coordination][T#1] master node [{node_t2}{tN0GU0iORmeSB6p8dVJ6nA}{H2TVjHoKTaO87YM9yTDKQg}{node_t2}{127.0.0.1}{127.0.0.1:25065}{ilmrt}{9.5.0}{8000099-9095000}] disconnected, restarting discovery [[node_t2][127.0.0.1:25065][internal:coordination/fault_detection/leader_check] disconnected]	
[2026-04-16T11:16:51,414][INFO ][o.e.c.s.ClusterApplierService][node_t0][clusterApplierService#updateTask][T#1] master node changed {previous [{node_t2}{tN0GU0iORmeSB6p8dVJ6nA}{H2TVjHoKTaO87YM9yTDKQg}{node_t2}{127.0.0.1}{127.0.0.1:25065}{ilmrt}{9.5.0}{8000099-9095000}], current []}, term: 1, version: 46, reason: becoming candidate: onLeaderFailure	
[2026-04-16T11:16:51,414][INFO ][o.e.c.s.ClusterApplierService][node_t1][clusterApplierService#updateTask][T#1] master node changed {previous [{node_t2}{tN0GU0iORmeSB6p8dVJ6nA}{H2TVjHoKTaO87YM9yTDKQg}{node_t2}{127.0.0.1}{127.0.0.1:25065}{ilmrt}{9.5.0}{8000099-9095000}], current []}, term: 1, version: 46, reason: becoming candidate: onLeaderFailure
```

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -331,9 +331,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeApproximationIT
   method: test {csv-spec:inlinestats.filterSometimesMatches}
   issue: https://github.com/elastic/elasticsearch/issues/146545
-- class: org.elasticsearch.versioning.ConcurrentSeqNoVersioningIT
-  method: testSeqNoCASLinearizability
-  issue: https://github.com/elastic/elasticsearch/issues/146568
 - class: org.elasticsearch.xpack.esql.qa.multi_node.GenerativeUnmappedLoadIT
   method: test
   issue: https://github.com/elastic/elasticsearch/issues/145900

--- a/server/src/internalClusterTest/java/org/elasticsearch/versioning/ConcurrentSeqNoVersioningIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/versioning/ConcurrentSeqNoVersioningIT.java
@@ -12,9 +12,13 @@ import org.apache.logging.log4j.Level;
 import org.elasticsearch.ExceptionsHelper;
 import org.elasticsearch.action.DocWriteResponse;
 import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.support.PlainActionFuture;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.ClusterStateUpdateTask;
 import org.elasticsearch.cluster.coordination.LinearizabilityChecker;
 import org.elasticsearch.cluster.coordination.LinearizabilityChecker.LinearizabilityCheckAborted;
 import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.service.ClusterService;
 import org.elasticsearch.common.ReferenceDocs;
 import org.elasticsearch.common.io.stream.InputStreamStreamInput;
 import org.elasticsearch.common.io.stream.NamedWriteable;
@@ -119,7 +123,7 @@ public class ConcurrentSeqNoVersioningIT extends AbstractDisruptionTestCase {
     // multiple threads doing CAS updates.
     // Wait up to 1 minute (+10s in thread to ensure it does not time out) for threads to complete previous round before initiating next
     // round.
-    public void testSeqNoCASLinearizability() {
+    public void testSeqNoCASLinearizability() throws Exception {
         final int disruptTimeSeconds = scaledRandomIntBetween(1, 8);
 
         assertAcked(
@@ -182,6 +186,31 @@ public class ConcurrentSeqNoVersioningIT extends AbstractDisruptionTestCase {
         }
 
         partitions.forEach(Partition::assertLinearizable);
+
+        // ensure master election has stabilized after disruption
+        assertBusy(ConcurrentSeqNoVersioningIT::publishTrivialClusterStateUpdate);
+    }
+
+    private static void publishTrivialClusterStateUpdate() {
+        var processed = new PlainActionFuture<>();
+        internalCluster().getCurrentMasterNodeInstance(ClusterService.class)
+            .submitUnbatchedStateUpdateTask("trivial cluster state update", new ClusterStateUpdateTask() {
+                @Override
+                public ClusterState execute(ClusterState currentState) {
+                    return ClusterState.builder(currentState).build();
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    processed.onFailure(e);
+                }
+
+                @Override
+                public void clusterStateProcessed(ClusterState oldState, ClusterState newState) {
+                    processed.onResponse(null);
+                }
+            });
+        safeGet(processed);
     }
 
     private class CASUpdateThread extends Thread {

